code/hud/hudets.cpp

```diff
@@ -117,6 +118,31 @@ void update_ets(object* objp, float fl_frametime)
 		}
 	}
 
+	// update max speed, moved from 'update_ets' function as part of max speed cleanup -Asteroth and wookieejedi
+	// AL 11-15-97: Rules for engine strength affecting max speed:
+	//						1. if strength >= 0.5 no affect
+	//						2. if strength < 0.5 then max_speed = sqrt(strength)
+	//
+	//					 This will translate to 71% max speed at 50% engines, and 31% max speed at 10% engines
+	//
+
+	float eng_current_strength = ship_get_subsystem_strength(ship_p, SUBSYSTEM_ENGINE);
+
+	// only update max speed if aggregate engine health has changed
+	// which helps minimze amount of overrides to max speed
+	if (eng_current_strength != ship_p->prev_engine_aggregate_hits) {
```

Comment:
Minor, but since we're going to compare this to `strength` it should probably be named as such instead of `..._hits` since we aren't using direct hp values.

Reply:
Good point, just updated